tests/Serializer/SerializerTest.php

```diff
@@ -96,6 +99,47 @@ public function testNull(bool $serializeAllObjects): void
         $this->assertNull($result);
     }
 
+    public function testRegisteredObjectSerializers(): void
+    {
+        $serializer = $this->createSerializer(new Options([
+            'class_serializers' => [
+                StubObject::class => $customSerializerCallback = static function (StubObject $object): array {
+                    return [
+                        'purpose' => $object->getPurpose(),
+                    ];
+                },
+            ],
+        ]));
+
+        $object = new StubObject();
+
+        $this->assertEquals([
+            'class' => \get_class($object),
+            'data' => $customSerializerCallback($object),
+        ], $this->invokeSerialization($serializer, $object));
+    }
+
+    public function testSerializableObject(): void
+    {
+        $serializer = $this->createSerializer();
+
+        $object = new StubSerializableInterfaceObject();
+
+        $this->assertEquals([
+            'class' => \get_class($object),
+            'data' => $object->toSentry(),
+        ], $this->invokeSerialization($serializer, $object));
+    }
+
+    public function testSerializableObjectThatThrowsAnException(): void
+    {
+        $serializer = $this->createSerializer();
+
+        $object = new StubSerializableInterfaceObjectThrowingException();
```

Comment:
Instead of a real object cannot we simply mock the interface and its methods as always?